resident/resident-service/src/main/java/io/mosip/resident/service/impl/ResidentVidServiceImpl.java

```diff
@@ -293,10 +294,12 @@ public Tuple2<ResponseWrapper<VidResponseDto>, String> generateVidV2(BaseVidRequ
 			notifyVidCreationFailureAndThrowException(requestDto, isV2Request, notificationRequestDto, eventId, residentTransactionEntity, e);
 		} finally {
 			if (Utility.isSecureSession() && residentTransactionEntity != null) {
-				//if the status code will come as null, it will set it as failed.
-				if(residentTransactionEntity.getStatusCode()==null) {
+				//if the status code or request summary will come as null, it will set it as failed.
+				if(residentTransactionEntity.getStatusCode() == null || residentTransactionEntity.getRequestSummary() == null) {
```

Comment:
Why this OR condition needed here?

Reply:
request summary is also mandatory, so if it will be missed then it will assign it here